src/test/java/uk/gov/companieshouse/documentsigningapi/signing/SigningServiceTest.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.companieshouse.documentsigningapi.signing;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import uk.gov.companieshouse.documentsigningapi.exception.DocumentSigningException;
+import uk.gov.companieshouse.documentsigningapi.exception.DocumentUnavailableException;
+import uk.gov.companieshouse.documentsigningapi.logging.LoggingUtils;
+
+@ExtendWith(MockitoExtension.class)
+public class SigningServiceTest {
+
+    @Mock
+    private LoggingUtils logger;
+
+    @Mock
+    private ResponseInputStream<GetObjectResponse> response;
+
+    @InjectMocks
+    private SigningService signingService = new SigningService("jks", "src/test/resources/keystore.jks", "testkey", "alias", logger);
+
+    @Test
+    @DisplayName("Throws DocumentUnavailableException when unable to load keystore")
+    void throwsDocumentUnavailableExceptionExceptionWhereHostIsNull() {
+        final DocumentUnavailableException exception = assertThrows(DocumentUnavailableException.class,
+            () -> signingService.signPDF(response));
+        assertThat(exception.getMessage(),
+            is("Unable to load Keystore or Certificate"));
+    }
+
+    @Test
+    @DisplayName("Throws DocumentSigningException when failing to obtain keystore")
+    void throwsDocumentSigningExceptionExceptionWhereHostIsNull() {
+        SigningService incorrectKeystoreTypeInitialised = new SigningService("unknown", "src/test/resources/keystore.jks", "testkey", "alias", logger);
+        final DocumentSigningException exception = assertThrows(DocumentSigningException.class,
+            () -> incorrectKeystoreTypeInitialised.signPDF(response));
+        assertThat(exception.getMessage(),
+            is("Failed to obtain proper KeyStore or Certificate"));
+    }
+}
```

Comment:
There appear to be 2 tests for failing to load or obtain the keystore, but no positive scenario tests. Is the intention to only test the positive scenario through the integration tests?

Reply:
Yes, due to the difficulty of testing the positive scenario (working with the keystores) and the coverage being already provided I chose to only the fails